server/resp/src/main/java/org/infinispan/server/resp/Resp3Handler.java

```diff
@@ -0,0 +1,305 @@
+package org.infinispan.server.resp;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.logging.LogFactory;
+import org.infinispan.commons.util.Version;
+import org.infinispan.server.core.logging.Log;
+import org.infinispan.util.concurrent.AggregateCompletionStage;
+import org.infinispan.util.concurrent.CompletionStages;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.util.CharsetUtil;
+
+public class Resp3Handler implements RequestHandler {
+   private static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass(), Log.class);
+
+   private static final Resp3Handler singleton = new Resp3Handler();
+
+   public static RequestHandler getInstance() {
+      return singleton;
+   }
+
+   private Resp3Handler() { }
+
+   @Override
+   public RequestHandler handleRequest(ChannelHandlerContext ctx, Cache<byte[], byte[]> cache, String type,
+                             List<byte[]> arguments) {
+      switch (type) {
+         case "HELLO":
+            byte[] respProtocolBytes = arguments.get(0);
+            String version = new String(respProtocolBytes, CharsetUtil.UTF_8);
+            if (!version.equals("3")) {
+               ctx.writeAndFlush(stringToByteBuf("-NOPROTO sorry this protocol version is not supported", ctx.alloc()));
+               break;
+            }
+
+            // TODO: need to do auth
+            String versionString = Version.getBrandVersion();
+            ctx.writeAndFlush(stringToByteBuf("%7\r\n" +
+                  "$6\r\nserver\r\n$15\r\nInfinispan RESP\r\n" +
+                  "$7\r\nversion\r\n$" + versionString.length() + "\r\n" + versionString + "\r\n" +
+                  "$5\r\nproto\r\n:3\r\n" +
+                  "$2\r\nid\r\n:184\r\n" +
+                  "$4\r\nmode\r\n$7\r\ncluster\r\n" +
+                  "$4\r\nrole\r\n$6\r\nmaster\r\n" +
+                  "$7\r\nmodules\r\n*0\r\n", ctx.alloc()));
+            break;
+         case "AUTH":
+            // TODO: do this
+            ctx.writeAndFlush(stringToByteBuf("-ERR not implemented yet" + "\r\n", ctx.alloc()));
```

Comment:
the interface `RequestHandler` could have a default `handleNotImplemented(ctx)`  method which invokes
`ctx.writeAndFlush(stringToByteBuf("-ERR not implemented yet\r\n", ctx.alloc()));`

Reply:
This is implemented in another PR that Tristan worked on that I will pull in today.